plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergSystemTables.java

```diff
@@ -190,4 +192,39 @@ public void testFilesTable()
                         "('equality_ids', 'array(integer)', '', '')");
         assertQuerySucceeds("SELECT * FROM test_schema.\"test_table$files\"");
     }
+
+    @Test
+    public void testAllFilesTable()
+    {
+        assertQuery("SHOW COLUMNS FROM test_schema.\"test_table$all_files\"",
+                "VALUES ('content', 'integer', '', '')," +
+                        "('file_path', 'varchar', '', '')," +
+                        "('file_format', 'varchar', '', '')," +
+                        "('record_count', 'bigint', '', '')," +
+                        "('file_size_in_bytes', 'bigint', '', '')," +
+                        "('column_sizes', 'map(integer, bigint)', '', '')," +
+                        "('value_counts', 'map(integer, bigint)', '', '')," +
+                        "('null_value_counts', 'map(integer, bigint)', '', '')," +
+                        "('nan_value_counts', 'map(integer, bigint)', '', '')," +
+                        "('lower_bounds', 'map(integer, varchar)', '', '')," +
+                        "('upper_bounds', 'map(integer, varchar)', '', '')," +
+                        "('key_metadata', 'varbinary', '', '')," +
+                        "('split_offsets', 'array(bigint)', '', '')," +
+                        "('equality_ids', 'array(integer)', '', '')");
+        assertQuerySucceeds("SELECT * FROM test_schema.\"test_table$all_files\"");
+
+        List<String> allFilesPath = computeActual("SELECT * FROM test_schema.\"test_table$all_files\"").
+                getMaterializedRows().stream().map(materializedRow -> (String) materializedRow.getField(1))
+                .collect(Collectors.toUnmodifiableList());
+
+        List<String> filesPath = computeActual("SELECT * FROM test_schema.\"test_table$files\"")
+                .getMaterializedRows().stream().map(materializedRow -> (String) materializedRow.getField(1))
+                .collect(Collectors.toUnmodifiableList());
+
+        // All files will contain duplicate data-file entries because there are 3 snapshots now.
+        // and each snapshot will refer to the data files.
+        assertEquals(true, allFilesPath.size() > filesPath.size());
+
+        assertEquals(true, allFilesPath.containsAll(filesPath));
```

Comment:
besides the existing assertions, could you please add some assertions on the field values that don't change from test to test (e.g. : lower_bounds, upper_bounds, file_format, record_count, ...) ? 

In a different commit you could do the same for the $files table.

Reply:
```
        // file format
        assertEquals("ORC", allFilesResult.get(0).getField(2));
        // row count
        assertEquals(2, allFilesResult.get(0).getField(3));
        // lower bound
        assertEquals(Map.of(1, "0", 2, "2019-09-08"), allFilesResult.get(0).getField(9));
        // upper_bound
        assertEquals(Map.of(1, "0", 2, "2019-09-08"), allFilesResult.get(0).getField(10));

```

we can add something like this?

files and all_files will return multiple files and can have different record count etc.

Reply:
added assertions @findinpath